VAMobile/src/screens/HealthScreen/Appointments/PastAppointments/PastAppointments.tsx

```diff
@@ -171,6 +165,20 @@ function PastAppointments({
   const pickerOptions = getPickerOptions()
   const [datePickerOption, setDatePickerOption] = useState(pickerOptions[0])
 
+  const filteredAppointments = useMemo(
+    () =>
+      datePickerOption.timeFrame === TimeFrameTypeConstants.PAST_THREE_MONTHS ||
+      datePickerOption.timeFrame === TimeFrameTypeConstants.PAST_ALL_CURRENT_YEAR
+        ? filterAppointments(appointmentsData?.data || [], true)
+        : appointmentsData?.data,
+    [appointmentsData?.data, datePickerOption],
+  )
+
+  useEffect(() => {
```

Comment:
nit: I’d consider replacing this useEffect with useMemo. While both only recalculate when dependencies change, useMemo avoids the extra render cycle caused by useEffect (one for the initial render, and another after the useEffect updates the state). This can lead to better performance and a cleaner render cycle.

```
const appointmentsToShow = useMemo(
  () => filteredAppointments?.slice((page - 1) * perPage, page * perPage) || [],
  [filteredAppointments, page, perPage]
)
```

Reply:
Changed to that